src/optimization/OptiSys.h

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) TrajoptLib contributors
+
+#pragma once
+
+template <typename Expr>
+concept ExprSys = requires(Expr expr, const Expr constExpr, double num) {
+  Expr();
+  Expr(num);
+  Expr(constExpr);
+  expr = expr;
+
+  -constExpr;
+  constExpr + constExpr;
+  constExpr - constExpr;
+  constExpr* constExpr;
+  constExpr / constExpr;
+
+  expr += constExpr;
+
+  expr = sin(constExpr);  // NOLINT
+  expr = cos(constExpr);  // NOLINT
+
+  expr = fmin(constExpr, constExpr);  // NOLINT
+  expr = fmax(constExpr, constExpr);  // NOLINT
+};                                    // NOLINT
```

Comment:
Is this last NOLINT required?

Reply:
Yes. If you don't put it there, it will become `std::fmin` which is not correct in this context.